### ISSUE TITLE
Make DigiToRawPreMixing depend on DigiToRaw

### ISSUE THREAD

**Issue by perrotta** [cms-sw/cmssw#21704]
```Configuration/StandardSequences/python/DigiToRawPreMixing_cff.py``` is now almost a clone of ```Configuration/StandardSequences/python/DigiToRaw_cff.py```

With such a design, every time someone has to update one of them (e.g. for era based customizations), one has to repeat exactly the same in the other code, with the non negligible risk of making mistakes.

I think that a much safer and cleaner solution could be that DigiToRawPreMixing_cff.py just import DigiToRaw from Configuration/StandardSequences/python/DigiToRaw_cff.py and only apply the needed customization that differentiate the two sequences.

At the moment, the only differences are the following:
- Remove ```L1TDigiToRaw``` from the DigiToRaw sequence in DigiToRawPreMixing
- Modify the flag ```hcalRawDatauHTR.premix = cms.bool(True)``` in DigiToRawPreMixing

As soon as #21681 (which includes some further parallel modifications to these configs) gets merged I will go ahead with proposing a simplification.

**Comment by perrotta** [comment]
assign reconstruction, simulation

**Comment by Dr15Jones** [comment]
Maybe this means we should control this via Eras instead of via imports?

**Comment by perrotta** [comment]
@Dr15Jones : that could be another option, less simple to me, but I can delegate :-)

By the way, the only place where DigiToRawPreMixing is accessed is in
http://cmslxr.fnal.gov/source/Configuration/Applications/python/ConfigBuilder.py#0959
(and right above there is also a DigiToRawDM_cff.py which is also customized via imports)

What is the suggestion from the framework team?

**Comment by davidlange6** [comment]
I cast a vote against eras. i had asked for a similar consolidation in raw2digi that worked well- i guess digi2raw should be straightforward to merge and simplify

**Comment by Dr15Jones** [comment]
If David doesn't want eras, then no eras. 
@davidlange6 for completion, what is your reason for not using eras here?

**Comment by makortel** [comment]
Adding @kpedro88 since we've talked with @mdhildreth about using eras for premixing customizations.

**Comment by davidlange6** [comment]
lets rephrase..

if you want to use eras for just this one file thus meaning that someone using premixing has to remember both an era and the right set of imports (or steps in cmsDriver), you've made things more complex for no reason

if instead you want to rewrite all the premixing configs such that a user just needs to --era premixing and  otherwise has the same config as for normal mixing DIGI steps, that sounds like an interesting possibility (even if it sounds like a lot of work).

in the second case - I would suggest using --processModifiers instead (which are the same thing with a more generic name as not to imply time-dependence which was the original use case behind "eras'.

**Comment by kpedro88** [comment]
> if instead you want to rewrite all the premixing configs such that a user just needs to --era premixing and otherwise has the same config as for normal mixing DIGI steps, that sounds like an interesting possibility (even if it sounds like a lot of work).

That would be my preference. (I didn't know a separate category of ProcessModifiers had been defined, but indeed it's there now: https://github.com/cms-sw/cmssw/tree/master/Configuration/ProcessModifiers/python)

**Comment by makortel** [comment]
@perrotta 
> As soon as #21681 (which includes some further parallel modifications to these configs) gets merged I will go ahead with proposing a simplification.

In my private developer area I have already moved `DigiToRawPreMixing` customizations to be done with a `cms.Modifier` in `DigiToRaw`. I'll make a PR after #22166 gets merged (it can wait #21681 as well if that won't require too much time to converge).

**Comment by perrotta** [comment]
Thank you @makortel !
I think that you can go ahead with your PR as soon as #22166 gets merged:  #21681 still needs some work, and (a tiny) part of that work could be to rebase it once your new PR will get also merged.

**Comment by makortel** [comment]
Addessed by #22210.

**Comment by makortel** [comment]
#22210 was merged so this issue can be closed now.

**Comment by perrotta** [comment]
+1
(Thank you Matti)
